### ISSUE TITLE
boot call is broken in 2.8.1

### ISSUE THREAD

**Issue by alexander-yakushev** [boot-clj/boot#707]
Works in 2.7.2:

```
$ boot call -e '(+ 2 2)' -p
4
```

Broken in 2.8.1:

```
$ boot call -e '(+ 1 2)' -p
  clojure.lang.ExceptionInfo: java.lang.Character cannot be cast to java.lang.String
    line: 65
java.lang.ClassCastException: java.lang.Character cannot be cast to java.lang.String
                          clojure.core/read-string                 core.clj: 3771
                              clojure.core/fnil/fn                 core.clj: 6486
                               boot.cli/parse-type                  cli.clj:   72
                               clojure.core/map/fn                 core.clj: 2752
                                               ...
                                  clojure.core/seq                 core.clj:  137
                 clojure.core.protocols/seq-reduce            protocols.clj:   24
                         clojure.core.protocols/fn            protocols.clj:   75
```

`boot call -f` throws the same error too which seems to relate to parsing symbol and form arguments.

**Comment by seancorfield** [comment]
That commit breaks Multi-Options per the documentation (set _or vectors_ are documented to be supported but vectors no longer work). This breaks the `run` task in all `boot/new` application projects:
```
(deftask run
  "Run the project."
  [a args ARG [str] "the arguments for the application."]
  (with-pass-thru fs
    (require '[taskarg.core :as app])
    (apply (resolve 'app/-main) args)))
```
Order *must* be preserved here.

This has also broken a number of tasks in our build pipeline at work where we use `[str]` because order is important.

**Comment by Raffinate** [comment]
I faced with an error when I tried to use boot new tenzing template to create new app project. I got interested why it doesn't work and find the commit and changes which caused that. After that I search open issues to boot and find this one, which happens to be exact same issue why tenzing template doesn't work. 

I tried to understand how this should be fixed, but wiki https://github.com/boot-clj/boot/wiki/Task-Options-DSL explicitly mentions that for sequential [...] optarg "if multiple arguments are supplied only there last is taken and no conjoining happens". So right now it's completely unclear for me how this issue should be fixed and how complex optargs should behave. It looks like "complex optargs" feature was not designed properly.

Is there a plan to overcome this situation already?
